pkg/upgrade/upgrade.go

```diff
@@ -78,29 +81,62 @@ func ManagedInstances(ctx context.Context, c client.Client, reader client.Reader
 }
 
 // ManagedInstance performs the necessary changes to bring the given Jaeger instance to the current version
-func ManagedInstance(ctx context.Context, client client.Client, jaeger v1.Jaeger) (v1.Jaeger, error) {
+func ManagedInstance(ctx context.Context, client client.Client, jaeger v1.Jaeger, latestVersion string) (v1.Jaeger, error) {
 	tracer := global.TraceProvider().GetTracer(v1.ReconciliationTracer)
 	ctx, span := tracer.Start(ctx, "ManagedInstance")
 	defer span.End()
+	currentSemVersion, _ := semver.NewVersion(jaeger.Status.Version)
+	startSemVersion, _ := semver.NewVersion("1.11.0")
+	latestSemVersion, _ := semver.NewVersion(latestVersion)
+
+	if currentSemVersion.LessThan(startSemVersion) {
+		// We don't know how to do an upgrade from versions lower than 1.11.0
+		jaeger.Logger().WithFields(log.Fields{
+			"instance":  jaeger.Name,
+			"namespace": jaeger.Namespace,
+			"to":        latestVersion,
+		}).Warn("Cannot automatically upgrade from versions lower than 1.11.0")
+		return jaeger, nil
+	}
+
+	versionLists := make([]*semver.Version, len(versions))
+	versionIndex := 0
+	for v := range versions {
+		v, err := semver.NewVersion(v)
```

Comment:
Shouldn't use the same variable name, as we still need access to the original version text. Possibly call this one `sm` or `semv`?